src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -101,13 +101,17 @@ public Checksum CreateChecksum(AnalyzerReference reference, CancellationToken ca
                         WriteUnresolvedAnalyzerReferenceTo(unresolved, writer);
                         break;
 
+                    case AnalyzerReference analyzerReference when analyzerReference.GetType().FullName == "Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioUnresolvedAnalyzerReference":
```

Comment:
found it. looks fine. but probably want to refactor this code so that we can have different service in VS layer which knows those vs layer only type. or make AnalyzerReference itself to implement checksum, WriteTo and ReadFrom API.